### ISSUE TITLE
no-render-return-value fires warning though I am not using the value

### ISSUE THREAD

**Issue by gndelia** [yannickcr/eslint-plugin-react#2037]
Basic piece of code to reproduce

```jsx
let newWrapper = document.createElement('div');
return new Promise(res => ReactDOM.render(<SomeComponent />, newWrapper, () => res(newWrapper.children[0])));
```

(I am doing this inside a map)

Running eslint gives the following error

>  error  Do not depend on the return value from ReactDOM.render  react/no-render-return-value

I am not using the value, I am just passing a function to Promise constructor. 

If I rewrote itlike this

```jsx
let newWrapper = document.createElement('div');
return new Promise(res => {
       ReactDOM.render(<SomeComponent />
}, newWrapper, () => res(newWrapper.children[0])));
```
where it is explicit that the value is not used, the error does not shown.

Is the first case a false positive? I understand it might be difficult to prove, but for Promisese (which are native) we can ensure the value is not being used.

Using

```
"eslint": "^4.18.2",
"eslint-plugin-react": "^7.7.0",
```

Note: I know this is not the way we should use React, it is just for a particular case inside my application.

**Comment by ljharb** [comment]
No, it’s not a false positive. Even if you aren’t using the return value, it’s still being passed into the promise.

The curly brace version is correct.

**Comment by gndelia** [comment]
I found a better solution than using an arrow function without return.
using [void operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void)

```javascript
new Promise(res => void ReactDOM.render(<Routes />, newWrapper, () => res(newWrapper.children[0])));
```

This now does not throw an error.

**Comment by ljharb** [comment]
That seems pretty silly when you could also do `// eslint-disable-next-line react/no-render-return-value`.

Closing as answered: https://github.com/yannickcr/eslint-plugin-react/issues/2037#issuecomment-435160552
